### DISCUSSION TITLE
Keycloak X mysql-connector-java-8.0.30 License Issue.

### DISCUSSION THREAD

**Discussion by ctgaurav** [keycloak/keycloak#25377] [Archive]
Hello, everyone,

I need your help. Currently, Keycloak X is delivered with default database drivers, and it comes with the built-in "keycloak-23.0.1\lib\lib\main\mysql.mysql-connector-java-8.0.30.jar." Unfortunately, this JAR file has a licensing issue as it is licensed under the GNU GPL. Therefore, unless we use the commercial version and pay for it, we should not include it in our deliveries.

To address this, we have replaced this driver with the MariaDB connector by configuring the driver update properties. However, when we try to remove the "mysql.mysql-connector-java-8.0.30.jar" file, Keycloak startup fails because it is referenced in the bootstrapping process of Quarkus. References to this JAR file are found in the following files:

- Binary file lib/lib/deployment/appmodel.dat matches
- Binary file lib/lib/deployment/deployment-class-path.dat matches
- Binary file lib/quarkus/quarkus-application.dat matches

Can anybody suggest how to completely remove the "mysql.mysql-connector-java-8.0.30.jar" from the Keycloak delivery without causing startup failures?

Your assistance is greatly appreciated.

**Comment by dipugit33** [comment]
Same here, have you got any assistance from anyone on this issue ?

**Comment by DL-YYA** [comment]
Anyone have a solution for this? Or can anyone tell me if using the built-in "mysql.mysql-connector-java-8.0.30.jar." needs to follow the gplv2 open source code when distributed?

**Comment by Snehalssg** [comment]
Anyone has any solution for this issue?

**Comment by stianst** [comment]
This is not the only GPL dependency included in Keycloak. Please elaborate on why this is a problem as there is no restrictions on re-distributing the mysql binary as long as the requirements of the license are followed.
